tools/debug_tool/static/pipeline_viewer.js

```diff
@@ -0,0 +1,236 @@
+function isIE() { return ((navigator.appName == 'Microsoft Internet Explorer') || ((navigator.appName == 'Netscape') && (new RegExp("Trident/.*rv:([0-9]{1,}[\.0-9]{0,})").exec(navigator.userAgent) != null))); }
+
+// Time between updating kotekan metrics
+const POLL_WAIT_TIME_MS = 1000;
+// Poll kotekan via web server every POLL_WAIT_TIME_MS and update page with new metrics
+function poll(label, index) {
+    get_data().then(function (new_buffers) {
+        update_buf_utl(new_buffers, label, index);
+    });
+
+    // Call poll() again to get the next message
+    setTimeout(() => { poll(label, index); }, POLL_WAIT_TIME_MS);
+}
+
+// Update buffer utilization
+function update_buf_utl(new_buffers, label, index){
+    for (var i of index) {
+        var name = d3.select(label._groups[0][i]).select("tspan").text();
+        d3.select(label._groups[0][i]).select("#utl")
+            .text(new_buffers[name].num_full_frame + "/" + new_buffers[name].num_frames);
+    }
+}
+
+// To avoid long string, break the label by '/'
+var insertLinebreaks = function (d, margin = 6) {
+    var el = d3.select(this);
+    var words = d.name.split('/');
+    var tspan_x = margin/2
+
+    // The first line will be used to check if it is a buffer
+    var tspan = el.append('tspan').text(words[0])
+                    .attr('x', tspan_x).attr('dy', '15')
+                    .attr("font-size", "15");
+
+    for (var i = 1; i < words.length; i++) {
+        tspan = el.append('tspan').text('/' + words[i]);
+        tspan.attr('x', tspan_x).attr('dy', '15')
+                .attr("font-size", "15");
+    }
+};
+
+// Read from endpoint /buffers to get buffer stats
+async function get_data() {
+    let response = await fetch("/buffers");
+
+    if (response.status == 502) {
+        // Status 502 is a connection timeout error,
+        // may happen when the connection was pending for too long,
+        // and the remote server or a proxy closed it
+        // let's reconnect
+    } else if (response.status != 200) {
+        // An error - let's show it
+        console.log("Error: " + response.statusText);
+        // Reconnect in one second
+        await new Promise(resolve => setTimeout(resolve, 1000));
+    } else {
+        // Get buffer stats
+        return await response.json();
+    }
+}
+
+class PipelineViewer {
+    #d3cola;
+    #svg;
+    #graph;
+    #link;
+    #node;
+    #label;
+
+    constructor(buffers, body) {
+        this.buffers = buffers;
+        this.bufNames = Object.keys(this.buffers);
+        this.body = body;
+    }
+
+    start_viewer() {
+        this.init_svg();
+        this.parse_data();
+        this.create_objs();
+        this.start_buff_ult();
+    }
+
+    init_svg(width = 960, height = 500) {;
```

Comment:
These optional parameters should be at the level of `start_viewer()` so that the user on the HTML side can adjust them if needed, same with `margin` on `create_objs()`